src/player.cpp

```diff
@@ -7350,7 +7354,25 @@ void player::hardcoded_effects(effect &it)
                 }
             }
         }
+    } else if (id == "grabbed"){
+        int intensity = 0;
+        for (int i = -1; i <= 1; i++) {
+            for (int j = -1; j <= 1; j++) {
+                tripoint dest( posx() + i, posy() + j, posz() );
+                if (g->mon_at(dest) != -1){
+                    intensity ++;
+                }
+            }
+        }
+        if (intensity > 0){
+            if (get_effect_int("grabbed") > intensity * 2){
```

Comment:
This formula looks wrong to me. If the intensity of your grabbed effect is higher than 2*zed number, the intensity is increased.